deploy-cloud-functions/action.yaml

```diff
@@ -0,0 +1,106 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Cloud Functions Deploy
+description: |-
+  Cloud Functions is a scalable Functions-as-a-Service (FaaS) to run your 
+  code with zero server management. Use this action to deploy your function or
+  update an existing Cloud Function.
+author: GoogleCloudPlatform
+
+inputs:
+  credentials:
+    description: |-
+      Optional service account key to use for authentication to GCS. This should be the JSON
+      formatted private key which can be exported from the Cloud Console. The
+      value can be raw or base64-encoded.
+    required: false
+    
+  name:
+    description: |-
+      Name of the Cloud Function.
+    required: true
+
+  description:
+    description: |-
+      Description for the CF.
+    required: false
+
+  project_id:
+    description: |-
+      Project to deploy the function in.
+    required: false
+
+  source_dir:
+    description: |-
+      Path to function deployment directory within the source repo.
+    required: false
+
+  env_vars:
+    description: |-
+      List of key-value pairs to set as environment variables in the form KEY1=VALUE1,KEY2=VALUE2.
+    required: false
+
+  entry_point:
+    description: |-
+      Name of a function (as defined in source code) that will be executed. Defaults to the resource name suffix, if not specified. 
+    required: false
+
+  runtime:
+    description: |-
+      Runtime to use for the function.
+    required: true
```

Comment:
Docs for `gcloud function deploy` mention that `runtime` is

> Required when deploying a new function; optional when updating an existing function.

Thanks for putting this together, it's very timely for me!

Reply:
Glad to hear that!
Yeah we required this just to be declarative. Is there a workflow that maybe impeded if it's always required?

Reply:
Not really, it just feels inconsistent next to the triggers which are similarly only required when creating a new function. I'm content if you leave this required.

Reply:
@xylo04 correct, we decided to just default to http as opposed to having another flag since `EventTrigger` object required some additional params.